packages/application-shell/src/components/custom-view-shell-authenticated/custom-view-shell-authenticated.tsx

```diff
@@ -1,23 +1,50 @@
 import { type ReactNode } from 'react';
 import { PageUnauthorized } from '@commercetools-frontend/application-components';
+import { entryPointUriPathToPermissionKeys } from '@commercetools-frontend/application-config/ssr';
 import { ApplicationContextProvider } from '@commercetools-frontend/application-shell-connectors';
-import type { ApplicationWindow } from '@commercetools-frontend/constants';
+import type {
+  ApplicationWindow,
+  CustomViewData,
+} from '@commercetools-frontend/constants';
 import {
   AsyncLocaleData,
   type TAsyncLocaleDataProps,
 } from '@commercetools-frontend/i18n';
+import { useIsAuthorized } from '@commercetools-frontend/permissions';
 import { ThemeProvider } from '@commercetools-uikit/design-system';
 import ApplicationLoader from '../application-loader';
 import { getBrowserLocale } from '../application-shell-provider/utils';
 import ConfigureIntlProvider from '../configure-intl-provider';
 import FetchProject from '../fetch-project';
 import FetchUser from '../fetch-user';
 
+type TCustomViewWithPermissionCheckProps = {
+  customViewId: string;
+  children?: ReactNode;
+};
+
+const CustomViewWithPermissionCheck = (
+  props: TCustomViewWithPermissionCheckProps
+) => {
+  const permissionKeys = entryPointUriPathToPermissionKeys(props.customViewId);
+
+  // Require View permission to render the application.
+  const canView = useIsAuthorized({
+    demandedPermissions: [permissionKeys.View],
+  });
```

Comment:
Currently this always yields `false`, until we adjust the applied permissions logic in the MC API to expose the computed permission keys for Custom Views (for local development).